### PR TITLE
Use temp unprotected list and user list for all features

### PR DESCRIPTION
**Pull request by SlayterDev**
Task/Issue URL: https://app.asana.com/0/414709148257752/1200749768481992/f
Tech Design URL:
CC:

**Description**:
This PR checks the temp unprotected list and user unprotected list for all privacy features.

**Steps to test this PR**:
1. Visit https://privacy-test-pages.glitch.me/privacy-protections/fingerprinting/
2. Run the test `Include all headers and all properties of window, navigator and screen objects` checked
3. `screen.pixelDepth` should be 24
4. Turn off protections
5. Reload and run tests again
6. `screen.pixelDepth` should be 32 (on iPhone 13 sim)
7. Edit `ios-config.json` and add this to the `https` exceptions list:
```
                {
                    "domain": "good.third-party.site",
                    "reason": "sitebreakage"
                }
```
8. In `PrivacyConfigurationManager.swift` comment out lines 79-81 to force load the embedded config
9. Visit http://privacy-test-pages.glitch.me/privacy-protections/https-upgrades/
10. Run the test. Upgrade navigation should show an http url. NOT https.

**Device Testing**:

* [x] iPhone SE (1st Gen)
* [x] iPhone 8
* [x] iPhone X
* [x] iPad

**OS Testing**:

* [x] iOS 13
* [x] iOS 14
* [x] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: Core/FingerprintUserScript.swift [hunk 1]
```diff
@@ -42,11 +42,17 @@ public class FingerprintUserScript: NSObject, UserScript {
         let screenSizeExceptions = PrivacyConfigurationManager.shared.privacyConfig
             .exceptionsList(forFeature: .fingerprintingScreenSize).joined(separator: "\n")
         
+        let tempExceptions = PrivacyConfigurationManager.shared.privacyConfig
+            .tempUnprotectedDomains.joined(separator: "\n")
+                + "\n"
+        + PrivacyConfigurationManager.shared.privacyConfig.userUnprotectedDomains.joined(separator: "\n")
```
Comment (bwaresiak) [inline Core/FingerprintUserScript.swift]: Shouldn't user unprotected domains be matched exactly, but temp unprotected and feature exceptions with wildcards (subdomains)?

File: DuckDuckGoTests/WebViewTestHelper.swift [hunk 2]
```diff
@@ -123,11 +123,14 @@ class WebKitTestHelper {
     static func preparePrivacyConfig(locallyUnprotected: [String],
                                      tempUnprotected: [String],
                                      contentBlockingEnabled: Bool,
-                                     exceptions: [String]) -> PrivacyConfiguration {
+                                     exceptions: [String],
+                                     httpsEnabled: Bool = false) -> PrivacyConfiguration {
```
Comment (bwaresiak) [inline DuckDuckGoTests/WebViewTestHelper.swift]: `httpsEnabled` feels unclear and can be misunderstood, maybe `httpsUpgradesEnabled`?

File: Core/PrivacyConfiguration.swift [hunk 3]
```diff
@@ -76,6 +76,7 @@ public enum PrivacyFeature: String {
     case fingerprintingBattery
     case fingerprintingScreenSize
     case gpc
+    case https
```
Comment (bwaresiak) [inline Core/PrivacyConfiguration.swift]: Maybe `httpsUpgrade` to be more explicit?

File: DuckDuckGoTests/HTTPSUpgradeTests.swift [hunk 4]
```diff
@@ -64,7 +64,13 @@ class HTTPSUpgradeTests: XCTestCase {
...
         let url = URL(string: "http://upgradable.url")!
 
-        let testee = HTTPSUpgrade(store: MockHTTPSUpgradeStore(bloomFilter: bloomFilter()))
+        let testee = HTTPSUpgrade(store: MockHTTPSUpgradeStore(bloomFilter: bloomFilter()),
+                                  privacyConfig: WebKitTestHelper.preparePrivacyConfig(
+                                    locallyUnprotected: [],
+                                    tempUnprotected: [],
+                                    contentBlockingEnabled: true,
+                                    exceptions: [],
+                                    httpsEnabled: true))
```
Comment (bwaresiak) [inline DuckDuckGoTests/HTTPSUpgradeTests.swift]: Besides this, shall we add a test for when feature is disabled?

**Comment by bwaresiak** [comment]
@SlayterDev Note, that I've merged #984 and it caused some conflicts.

File: Core/fingerprint.js [hunk 5]
```diff
@@ -180,11 +180,26 @@
...
+    if (userExcluded) {
+        return;
+    }
+    
     // walk up the domain to see if it's unprotected
     while (domainParts && domainParts.length > 1) {
         let partialDomain = domainParts.join(".")
+        
+        excludeAll = `
+                ${tempUnprotectedDomains}
```
Comment (jonathanKingston) [inline Core/fingerprint.js]: In the ESLint change for iOS we agreed to make this to `$TEMP_UNPROTECTED_DOMAINS$` and all the other corresponding ones the same format too.

File: Core/PrivacyConfiguration.swift (hunk 3 above)
Comment (SlayterDev) [reply Core/PrivacyConfiguration.swift]: This matches the feature in the config. But I can change the case name and set a raw string if its too unclear.

File: Core/fingerprint.js (hunk 5 above)
Comment (SlayterDev) [reply Core/fingerprint.js]: Yes that _just_ landed so I'll be having to update these anyway.

Comment (jonathanKingston) [reply Core/fingerprint.js]: Course, it was a FYI not error report :)

File: Core/PrivacyConfiguration.swift (hunk 3 above)
Comment (bwaresiak) [reply Core/PrivacyConfiguration.swift]: Yes, please - I'd prefer to be explicit.

**Review by bwaresiak — APPROVED** [review]
Great work!

**Pull request merged**